pkg/aws/aws.go

```diff
@@ -177,8 +183,103 @@ func (a *AwsClient) listInstancesWithFilter(filters []*ec2.Filter) ([]*ec2.Insta
 	return instances, nil
 }
 
+func (a Client) ListStopEventsForSpecificInstances(instances []*ec2.Instance) ([]*cloudtrail.Event, error) {
+	const backoffupperlimit = 5 * time.Minute
+	backoffoptions := wait.Backoff{
+		Duration: 2 * time.Second,
+		Factor:   2,
+		Cap:      backoffupperlimit,
+	}
+
+	events := []*cloudtrail.Event{}
+
+	idToStopTime, err := populateIdToStopTimeMap(instances)
+	if err != nil {
+		return nil, fmt.Errorf("could not populate the idToCloudtrailEvent map: %w", err)
+	}
+
+	err = wait.ExponentialBackoff(backoffoptions, func() (bool, error) {
+		idToCloudtrailEvent := make(map[string]*cloudtrail.Event)
```

Comment:
I would put this outside of the backoff retry, as we are by allocating new memory on every retry. And not sure, how fast garbage collection will be.